packages/playwright-core/src/server/trace/recorder/snapshotterInjected.ts

```diff
@@ -438,13 +438,13 @@ export function frameSnapshotStreamer(snapshotStreamer: string, removeNoScript:
             expectValue(value);
             attrs[kSelectedAttribute] = value;
           }
-          if (nodeName === 'CANVAS') {
-            const boundingRect = (element as HTMLCanvasElement).getBoundingClientRect();
+          if (nodeName === 'CANVAS' || nodeName === 'IFRAME' || nodeName === 'FRAME') {
+            const boundingRect = (element as HTMLElement).getBoundingClientRect();
             const value = JSON.stringify({
-              left: boundingRect.left / window.innerWidth,
```

Comment:
By moving this division to the rendering stage, we make an implicit assumption that `window.innerWidth` will be the same at snapshot time and at rendering time. I'm not sure that's true. Do you think we could perform all the heavy lifting at snapshot time instead? That way, we can be more sure that the bounding rect actually refers to the right pixels in the screenshot.

Reply:
I thought of keeping that at snapshot time, but the problem is that most likely we'll hit on XSS issues. I moved those computations to rendering stage because at that point we'll no longer have XSS issues (all rendered frames are under the same domain).

Nevertheless, the end result should be the same, because I capture `window.innerWidth` for each frame and save it in its `__playwright_bounding_rect__` attribute.

Reply:
What I said was not 100% correct: actually the `window.inner{Width,Height}` is not from iframe's `__playwright_bounding_rect__ `, it was already being captured at snapshot time:

https://github.com/microsoft/playwright/blob/d0debf67c0eab87238c51e86e3d9fbc70ca9a260/packages/playwright-core/src/server/trace/recorder/snapshotterInjected.ts#L597-L600

and those are the dimensions I use at rendering time. That's why I changed the `snapshotScript` signature to receive the viewport structure.